src/Command/Argument/Search/Arguments.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2023 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Command\Argument\Search;
+
+use InvalidArgumentException;
+use Predis\Command\Argument\ArrayableArgument;
+
+class Arguments implements ArrayableArgument
```

Comment:
Name it CreateArguments; you'd have naming leeway for search and aggregate in future.